apps/docs/features/docs/Reference.navigation.client.tsx

```diff
@@ -60,14 +58,33 @@ export function ReferenceNavigationScrollHandler({
   children,
   ...rest
 }: PropsWithChildren & HTMLAttributes<HTMLDivElement>) {
+  const parentRef = useRef<HTMLElement>()
   const ref = useRef<HTMLDivElement>()
   const initialScrollHappened = useContext(ReferenceContentInitiallyScrolledContext)
 
+  useEffect(() => {
+    if (!ref.current) return
+
+    let scrollingParent: HTMLElement = ref.current
+
+    while (scrollingParent && !(scrollingParent.scrollHeight > scrollingParent.clientHeight)) {
+      scrollingParent = scrollingParent.parentElement
+    }
+
+    parentRef.current = scrollingParent
+  }, [])
+
   const scrollActiveIntoView = useCallback(() => {
     const currentLink = ref.current?.querySelector('[aria-current=page]') as HTMLElement
     if (currentLink && !isElementInViewport(currentLink)) {
-      currentLink.scrollIntoView({
-        block: 'center',
+      // Calculate the offset of the current link relative to scrollingParent
+      // and scroll the parent to the top of the link.
+      const offsetTop = currentLink.offsetTop
+      const parentOffsetTop = parentRef.current?.offsetTop ?? 0
+      const scrollPosition = offsetTop - parentOffsetTop
+
+      parentRef.current?.scrollTo({
+        top: scrollPosition - 60 /* arbitrary padding */,
```

Comment:
**question**

If that's meant to offset the scrolling, doesn't that match the "top navbar" height (50px) + a bit of padding ?

Reply:
yep that's true, will update comment 👍🏼